src/ansys/sound/core/sound_composer/source_broadband_noise.py

```diff
@@ -0,0 +1,334 @@
+    """
+
+    def __init__(self, file: str = "", source_control: SourceControlTime = None):
+        """Class instantiation takes the following parameters.
+
+        Parameters
+        ----------
+        file : str, default: ""
+            Path to the broadband noise file. Supported files are text files with the header
+            `AnsysSound_BBN`. If left empty, the broadband noise source is not loaded.
+        source_control : SourceControlTime, default: None
+            Source control, consisting of the control parameter values over time, to use when
+            generating the sound from this source.
+        """
+        super().__init__()
+        self.source_control = source_control
+
+        # Define DPF Sound operators.
+        self.__operator_load = Operator(ID_COMPUTE_LOAD_SOURCE_BBN)
+        self.__operator_generate = Operator(ID_COMPUTE_GENERATE_SOUND_BBN)
+
+        if len(file) > 0:
+            self.load_source_bbn(file)
+        else:
+            self.source_bbn = None
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        # Source info.
+        if self.source_bbn is not None:
+            spectrum_type, delta_f, control_name, control_unit, control_values = (
+                self.__extract_bbn_info()
+            )
+
+            # Source name.
+            str_name = self.source_bbn.name
+            if str_name is None:
+                str_name = ""
+
+            # Spectrum type. TODO: reactivate if/else when quantity_type is available in python.
+            # if spectrum_type == "NARROWBAND":
+            #     str_type = f"{spectrum_type} (DeltaF: {delta_f:.1f} Hz)"
+            # else:
+            #     str_type = spectrum_type
+            str_type = spectrum_type
+
+            # Spectrum control values.
+            control_values = np.round(control_values, 1)
+            if len(control_values) > 10:
+                str_values = f"{str(control_values[:5])[:-1]} ... {str(control_values[-5:])[1:]}"
+            else:
+                str_values = str(control_values)
+
+            str_source = (
+                f"'{str_name}'\n"
+                f"\tSpectrum type: {str_type}\n"
+                f"\tSpectrum count: {len(self.source_bbn)}\n"
+                f"\tControl parameter: {control_name}, {control_unit}\n"
+                f"\t\t{str_values}"
+            )
+        else:
+            str_source = "Not set"
+
+        # Source control info.
+        if self.is_source_control_valid():
+            str_source_control = (
+                f"{self.source_control.control.name}\n"
+                f"\tMin: {self.source_control.control.data.min()}\n"
+                f"\tMax: {self.source_control.control.data.max()}\n"
+                f"\tDuration: "
+                f"{self.source_control.control.time_freq_support.time_frequencies.data[-1]} s"
+            )
+        else:
+            str_source_control = "Not set"
```

Comment:
@ansaminard for more coherence with SourceSpectrum, I would replace "Not set" by "Not set/valid"

Reply:
@anshlachamb Well in that situation, we were checking more than simply a set source control (namely that the duration was also >0). Here we only test whether it is set or not. I think the consistency of messages between classes matters less than the consistency between messages and what's actually being done. So I would leave it as is. It is more informative this way, for the user